### PR TITLE
Refactor guard clauses into their own class

### PR DESCRIPTION
**Pull request by matthias-ronge**
I was just reading this [article (in German) about Guard Clauses](https://www.heise.de/hintergrund/Fehlersuche-Guard-Clauses-den-Code-auf-Fehler-vorbereiten-9241426.html?seite=all) and I think it makes sense. It is extremely rare that such checks are already in our code. I moved the first tests into a separate class and refactored them from the code. My suggestion is to use them more regularly.

### CODE REVIEW

**Comment by matthias-ronge** [comment]
- moved class to `utils` package
- tests added
- yes, I am asking for opinions here @solth @BartChris @markusweigelt @oliver-stoehr 
- and yes, if we decide to use this, we should use it in way more places. This PR is just a first throw.

**Comment by matthias-ronge** [comment]
Even if the request for a generous implementation of Guards did not make it into the development fund, would you merge this anyway? Then, we could at least _start_ using them.

File: Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
+package org.kitodo.utils;
```
Comment (henning-gerhardt) [inline Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java:1]: Missing license header.

File: Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+import org.junit.Test;
+
+public class GuardTest {
```
Comment (henning-gerhardt) [inline Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java:14]: Some notes:
- write a single test method for a single test scenario and not all scenarios in one test method
- exception handling can be done in a lot more readable way. See https://www.baeldung.com/junit-assert-exception for Junit 5 and Junit 4 usage - where I prefer Junit 5

**Review by henning-gerhardt — CHANGES_REQUESTED** [review]

**Comment by matthias-ronge** [comment]
No idea why the build fails here. It works fine for me, both in Eclipse and in the shell:

    [INFO] Running org.kitodo.utils.GuardTest
    [INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.224 s -- in org.kitodo.utils.GuardTest

File: Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java (hunk 1 above)
Comment (matthias-ronge) [reply Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java:12]: License header inserted.

File: Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java (hunk 2 above)
Comment (matthias-ronge) [reply Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java:25]: - Tests divided.
- I used Junit 5. Yes, I recently discovered that there is this new variant. Much nicer!

File: Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+class GuardTest {
```
Comment (henning-gerhardt) [inline Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java:26]: Maybe change the visibility of the class and the methods to "public" - I got som warnings from CodeQL as the new test class is not in use.

**Review by henning-gerhardt — APPROVED** [review]
Only a small note - can may be ignored if CodeQL is not complaining.

File: Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    void isNotNullShouldFailForNullObjekt() {
+        IllegalArgumentException e = assertThrows(IllegalArgumentException.class, () -> {
+            Object object = null;
+            Guard.isNotNull("object", object);
+        }, "should throw IllegalArgumentException");
+        assertEquals("'object' must not be null", e.getMessage(), "should provide an understandable description");
+    }
+
+    @Test
+    @Tag("isPositive(double)")
```
Comment (stweil) [inline Kitodo-API/src/test/java/org/kitodo/utils/GuardTest.java:115]: This  and seven more similar lines cause warnings:
```
Jul 05, 2024 6:24:52 PM org.junit.jupiter.engine.descriptor.JupiterTestDescriptor lambda$getTags$1
WARNING: Configuration error: invalid tag syntax in @Tag("isPositive(double)") declaration on [void org.kitodo.utils.GuardTest.isPositiveDoubleShouldFailForZero()]. Tag will be ignored.
```

Should these lines be removed?

**Pull request merged**